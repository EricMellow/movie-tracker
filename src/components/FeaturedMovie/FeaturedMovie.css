.featuredMovie {
  width: 67%;
  height: 385px;
  background-size: cover;
  position: relative;
  text-align: left;
  border-radius: 5px;
}

.favoriteButton {
  width: 115px;
  height: 60px;
  border-radius: 0 0 5px 5px;
  background-color: #171717;
  position: absolute;
  top: 0;
  right: 30px;
  padding: 0 20px;
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.faveIcon {
  width: 27px;
  height: 27px;
  display: inline-block;
  background-size: cover;
  background-repeat: no-repeat;
  background-image: url('./favorite-icon-inactive.png');
}

.favorite .faveIcon {
  background-image: url('./favorite-icon-active.png');
}

.favoriteButton p {
    margin: 0;
    width: 75px;
    font-size: 0.95rem;
}

.favoriteButton:hover {
  cursor: pointer;
}

.movieOverview h2 {
  text-transform: uppercase;
  margin-top: 0;
}

.movieOverview {
  font-size: .8rem;
  width: 270px;
  height: 95px;
  border-radius: 5px 5px 0 0;
  background-color: #171717;
  position: absolute;
  bottom: 0;
  left: 45px;
  padding: 15px;
}

.movieOverview p {
  font-size: .75rem;
  line-height: 1rem;
  height: 58px;
  color: #ccc;
  margin-top: 0px;
  overflow: hidden;
  display: block;
  text-overflow: ellipsis;
}

.loginRequest {
  width: auto;
  background: #333;
  padding: 15px 20px 15px 20px;
  display: inline-block;
  border-radius: 5px 5px 0 0;
  position: absolute;
  top: -50px;
  right: 30px;
  color: #ffba00;
  box-shadow: inset 1px 1px 1px #000;
}

.movieOverview h2 {
  margin-bottom: 20px;
}

.noFavorites {
  width: auto;
  background: #111;
  padding: 15px 20px 15px 20px;
  display: inline-block;
  border-radius: 5px 5px 0 0;
  position: absolute;
  top: -50px;
  right: 0;
  color: #ffba00;
  box-shadow: inset 1px 1px 1px #000;
  position: relative;
  margin-top: 95px;
  height: 30px;
  padding-top: 25px;
  border-radius: 5px;
  display: -ms-flexbox;
  display: flex;
  align-content: center;
}