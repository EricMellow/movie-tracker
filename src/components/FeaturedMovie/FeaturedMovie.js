import React, { Component } from 'react';
import './FeaturedMovie.css';
import { connect } from 'react-redux';
import { addFavoriteMovie } from '../../actions/index'

class FeaturedMovie extends Component {

  handleFavoriteClick = () => {
    const selectedMovie = this.props.recentMovies.find(movie => {
      return movie.id === this.props.movieId;
    });
    this.props.addFavorite(selectedMovie);
    this.addFavoriteToDatabase(selectedMovie);
  }

  addFavoriteToDatabase = async (selectedMovie) => {
    const url = 'http://localhost:3000/api/users/favorites/new';
    await fetch(url, {
      method: 'POST',
      headers: {
        "Content-Type": 'application/json'
      },
      body: JSON.stringify({
        movie_id: selectedMovie.id,
        user_id: this.props.userId,
        title: selectedMovie.title,
        poster_path: selectedMovie.poster,
        release_date: selectedMovie.release,
        rating: selectedMovie.rating,
        overview: selectedMovie.overview,
        backdrop: selectedMovie.backdrop
      })
    })
  }

  render() {
    const featuredMovie = this.props.recentMovies.find(movie => {
      return movie.id === this.props.movieId;
    });
    if (featuredMovie) {
      const image = `https://image.tmdb.org/t/p/w1280${featuredMovie.backdrop}`;
      const background = { backgroundImage: `url( ${image} )` };
      const overview = featuredMovie.overview.substr(0, 300);

      return (
        <div className="featuredMovie" style={background} >
          <div className="favoriteButton" onClick={this.handleFavoriteClick}>
            <p>Add to Favorites</p>
          </div>
          <div className="movieOverview">
            <h2>{featuredMovie.title}</h2>
            <p>{overview} ...</p>

          </div>
        </div>
      );
    } else {
      return (
        <p>LOADING</p>);
    }

  }
}

const mapStateToProps = (state) => ({
  recentMovies: state.recentMovies,
  movieId: state.selectedMovieId,
  userId: state.userId
});

const mapDispatchToProps = (dispatch) => ({
  addFavorite: (selectedMovie) => dispatch(addFavoriteMovie(selectedMovie))
});

export default connect(mapStateToProps, mapDispatchToProps)(FeaturedMovie);


