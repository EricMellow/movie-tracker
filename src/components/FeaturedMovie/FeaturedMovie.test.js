import React from 'react';
import { FeaturedMovie, mapStateToProps, mapDispatchToProps } from "./FeaturedMovie";
import { shallow } from 'enzyme';


describe('Card', () => {
  let wrapper;
  let props;

  beforeEach(() => {
    props = {
      favoriteMovies: [{
        title: "Happy Days",
        movie_id: 12345,
        overview: 'string string string'
      }],
      movieId: 12345,
      recentMovies: [
        {
          title: "Happy Days",
          movie_id: 12345,
          overview: 'string string string'
        },
        {
          title: "Sad Days",
          movie_id: 23456,
          overview: 'string string string'
        }
      ]
    };

    wrapper = shallow(<FeaturedMovie {...props}/>);
  });

  it('should match the snapshot', () => {

    expect(wrapper).toMatchSnapshot();
  });

});