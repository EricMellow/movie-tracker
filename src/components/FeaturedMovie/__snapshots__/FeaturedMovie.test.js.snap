// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FeaturedMovie
    favoriteMovies={
        Array [
            Object {
              "movie_id": 12345,
              "overview": "string string string",
              "title": "Happy Days",
            },
          ]
    }
    movieId={12345}
    recentMovies={
        Array [
            Object {
              "movie_id": 12345,
              "overview": "string string string",
              "title": "Happy Days",
            },
            Object {
              "movie_id": 23456,
              "overview": "string string string",
              "title": "Sad Days",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        null,
        <div
          className="favoriteButton"
          onClick={[Function]}
>
          <p>
                    Remove Favorite
          </p>
          <div
                    className="faveIcon"
          />
</div>,
        <div
          className="movieOverview"
>
          <h2>
                    Happy Days
          </h2>
          <p>
                    string string string
                     ...
          </p>
</div>,
      ],
      "className": "featuredMovie favorite",
      "style": Object {
        "backgroundImage": "url( https://image.tmdb.org/t/p/w1280undefined )",
      },
    },
    "ref": null,
    "rendered": Array [
      null,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p>
              Remove Favorite
</p>,
            <div
              className="faveIcon"
/>,
          ],
          "className": "favoriteButton",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Remove Favorite",
            },
            "ref": null,
            "rendered": "Remove Favorite",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "faveIcon",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2>
              Happy Days
</h2>,
            <p>
              string string string
               ...
</p>,
          ],
          "className": "movieOverview",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Happy Days",
            },
            "ref": null,
            "rendered": "Happy Days",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "string string string",
                " ...",
              ],
            },
            "ref": null,
            "rendered": Array [
              "string string string",
              " ...",
            ],
            "type": "p",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          null,
          <div
            className="favoriteButton"
            onClick={[Function]}
>
            <p>
                        Remove Favorite
            </p>
            <div
                        className="faveIcon"
            />
</div>,
          <div
            className="movieOverview"
>
            <h2>
                        Happy Days
            </h2>
            <p>
                        string string string
                         ...
            </p>
</div>,
        ],
        "className": "featuredMovie favorite",
        "style": Object {
          "backgroundImage": "url( https://image.tmdb.org/t/p/w1280undefined )",
        },
      },
      "ref": null,
      "rendered": Array [
        null,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p>
                Remove Favorite
</p>,
              <div
                className="faveIcon"
/>,
            ],
            "className": "favoriteButton",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Remove Favorite",
              },
              "ref": null,
              "rendered": "Remove Favorite",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "faveIcon",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2>
                Happy Days
</h2>,
              <p>
                string string string
                 ...
</p>,
            ],
            "className": "movieOverview",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Happy Days",
              },
              "ref": null,
              "rendered": "Happy Days",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "string string string",
                  " ...",
                ],
              },
              "ref": null,
              "rendered": Array [
                "string string string",
                " ...",
              ],
              "type": "p",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
